### PR TITLE
gha: keep trusted and untrusted paths separate, and simplify actions reference

### PR DESCRIPTION
**Pull request by giorio94**
A few GHA workflows got recently modified to hardcode the repository and branch of the actions hosted locally (e.g., [1]). This was a security measure, as they are triggered after checking out the untrusted context (i.e., PR branch), and thus it would be possible for an external PR to inject malicious code.

Yet, at the same time, this change mostly defeats the smooth development process enabled by ariane (which automatically uses the workflow and context from the PR for trusted branches -- i.e., in cilium/cilium), requiring again to manually modify those references for testing purposes. Similarly, it also requires manual adaptations when changes are backported to stable branches, or to allow running them from forks, which are easy to overlook.

As an alternative solution, let's only check out the helm chart from the untrusted context in a separate directory, without overriding any of the trusted files (i.e., from the target branch) retrieved initially. This way, we are guaranteed that the local github actions are always trusted (as we are not overriding them, nor we are executing any script which could modify them), and can be invoked directly, without any additional constraint. A key aspect for this is that helm charts cannot execute arbitrary code in the client host.

Another difference, compared to the previous approach, is that now we also execute the `./contrib/scripts/kind.sh` script from the trusted context (i.e., target branch) instead of the PR context. However, this file is effectively part of the workflow definition, and this change brings consistency with the rest of it. The same also applies for the Gateway API conformance tests.

As an additional security measure, let's also postpone the checkout of the context after the setup of the test environment.

\[1\]: 654d92f29c4f ("ci-e2e: Use lvh-kind in secure way")

```release-note
Rework GHA workflows to checkout the untrusted context in a separate directory for increased separation
```

### CODE REVIEW

**Review by thorn3r — APPROVED** [review]
Nice work :+1:

**Review by youngnick — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

**Comment by giorio94** [comment]
Rebased onto main to fix conflicts.

**Review by pchaigno — APPROVED** [review]

File: .github/workflows/conformance-aks.yaml [hunk 1]
```diff
@@ -139,6 +139,7 @@ jobs:
         uses: ./.github/actions/helm-default
         with:
           image-tag: ${{ inputs.SHA }}
+          chart-dir: ./untrusted/install/kubernetes/cilium
```
Comment (joestringer) [inline .github/workflows/conformance-aks.yaml:142]: How does this pull from `untrusted/` if the `untrusted` path is only checked out on line 224?

Comment (giorio94) [reply .github/workflows/conformance-aks.yaml:142]: The `helm-default` action [only configures a variable](https://github.com/cilium/cilium/blob/c902a23a8f05162f783abbe870da7ff14e90551a/.github/actions/helm-default/action.yaml#L35), so it doesn't matter if the path has already been checked out at this point, as long as it gets checked out before the actual installation process.

**Pull request merged**